src/core/usecases/runningPackageConfigs.ts

```diff
@@ -0,0 +1,194 @@
+        )
+};
+
+export const thunks = {
+    "saveRunningPackageConfig":
+        (params: { runningPackageConfig: RunningPackageConfig }): ThunkAction =>
+        async (dispatch, getState) => {
+            const { runningPackageConfig } = params;
+            if (
+                isRunningPackageConfigInStore({
+                    "runningPackageConfigs":
+                        getState().runningPackageConfigs.runningPackageConfigs,
+                    runningPackageConfig: runningPackageConfig
+                })
+            ) {
+                return;
+            }
+
+            const runningPackageConfigWithSameName = getState()
+                .runningPackageConfigs.runningPackageConfigs.filter(
+                    ({ catalogId, packageName }) =>
+                        runningPackageConfig.catalogId === catalogId &&
+                        runningPackageConfig.packageName === packageName
+                )
+                .find(({ name }) => runningPackageConfig.name === name);
+
+            if (runningPackageConfigWithSameName !== undefined) {
+                dispatch(
+                    actions.runningPackageConfigDeleted({
+                        "serviceId": runningPackageConfigWithSameName.name
+                    })
+                );
+            }
+
+            dispatch(
+                actions.runningPackageConfigSaved({
+                    runningPackageConfig: runningPackageConfig
+                })
+            );
+            await dispatch(privateThunks.syncWithUserConfig());
+        },
+    "deleteRunningPackageConfig":
+        (params: { serviceId: string }): ThunkAction =>
+        async dispatch => {
+            const { serviceId } = params;
+
+            dispatch(
+                actions.runningPackageConfigDeleted({
+                    serviceId
+                })
+            );
+            await dispatch(privateThunks.syncWithUserConfig());
+        },
+    /** Pure */
+    "isRestorableRunningConfigInStore":
+        (params: {
+            runningPackageConfigs: RunningPackageConfig[];
+            runningPackageConfig: RunningPackageConfig;
+        }): ThunkAction<boolean> =>
+        () =>
+            isRunningPackageConfigInStore(params)
+};
+
+function isRunningPackageConfigInStore(params: {
+    runningPackageConfigs: RunningPackageConfig[];
+    runningPackageConfig: RunningPackageConfig;
+}) {
+    const { runningPackageConfig, runningPackageConfigs } = params;
+
+    return !!runningPackageConfigs.find(runningPackageConfig_i =>
+        areSameRunningPackageConfig(runningPackageConfig_i, runningPackageConfig)
+    );
+}
+
+function areSameRunningPackageConfig(
+    runningPackageConfiguration1: RunningPackageConfig,
+    runningPackageConfiguration2: RunningPackageConfig
+): boolean {
+    return runningPackageConfiguration1.name === runningPackageConfiguration2.name;
+}
+
+export const selectors = (() => {
+    function restorableRunnigPackageConfigs(rootState: State) {
```

Comment:
The thunk has been renamed but the selector hasn't.  
What's this selector? How does it differs from the internal state.  
If it's doesn't, just give it the same name.